content/blog/production-ready-object-detection-model-training-workflow-with-hpe-machine-learning-development-environment.md

```diff
@@ -0,0 +1,1107 @@
+    dataset_file: coco
+    backend: aws # specifiy the backend you want to use.  one of: gcs, aws, fake, local
+    data_dir: determined-ai-coco-dataset # bucket name if using gcs or aws, otherwise directory to dataset
+    masks: false
+    num_workers: 4
+
+    device: cuda
+environment:
+    environment_variables:                                                                          
+        - NCCL_DEBUG=INFO                                                                           
+        # You may need to modify this to match your network configuration.                          
+        - NCCL_SOCKET_IFNAME=ens,eth,ib
+bind_mounts:
+    - host_path: /tmp
+      container_path: /data
+      read_only: false
+scheduling_unit: 400
+# scheduling_unit: 40
+min_validation_period:
+    batches: 1288 # For Real training
+searcher:
+  name: grid
+  metric: mAP
+  smaller_is_better: false
+  max_length:
+    batches: 51520 # 50*1288 == 51520# Real Training
+
+records_per_epoch: 1288
+resources:
+    # slots_per_trial: 16
+    slots_per_trial: 8
+    shm_size: 2000000000
+max_restarts: 0
+
+entrypoint: python3 -m determined.launch.torch_distributed --trial model_def:ObjectDetectionTrial
+```
+
+```python
+# Run the below cell to run a hyperparameter search experiment
+!det e create determined_files/const-distributed-search.yaml determined_files/
+```
+
+```
+Preparing files to send to master... 312.2KB and 40 files
+Created experiment 79
+```
+
+## Load Checkpoint of Trained Experiment.
+
+Replace the `<EXP_ID>` and run the below cells with the experiment ID once the experiment is completed.
+
+```python
+from determined_files.utils.model import build_frcnn_model
+from utils import load_model_from_checkpoint
+from determined.experimental import Determined,client
+```
+
+```python
+experiment_id = 76
+MODEL_NAME = "xview-fasterrcnn"
+checkpoint = client.get_experiment(experiment_id).top_checkpoint(sort_by="mAP", smaller_is_better=False)
+print(checkpoint.uuid)
+loaded_model = load_model_from_checkpoint(checkpoint)
+```
+
+Now that we have a checkpoint from our trained object detection model, we can deploy it to Kserve to run inference and predictions.  
+
+# Part 5: Deploying Trained Model on Kserve
+
+This notebook walks you each step to deploy a custom object detection model on KServe. 
+
+We will show you how to:
+
+* Install Kserve Natively using Kind and Knative
+* Create a Persistent Volume Claim for local model deployment
+* Preparing custom model for Kserve inference
+* Deploying model using a KServe InferenceService
+* Complete a sample request and plot predictions
+
+Note: This notebook was tested on a Linux-based machine with Nvidia T4 GPUs. We also assume Docker is installed in your Linux system/environment
+
+Let's get started!
+
+## Pre-reqs: Setting up Python and Jupyter Lab environment
+
+Run the below commands to set up a python virtual environment, and install all the python packages needed for this tutorial
+
+```
```

Comment:
make sure to specify a syntax language (for example, bash, shell, or markdown) for the code snippet